org-roam-node.el

```diff
@@ -829,23 +829,50 @@ Any top level properties drawers are incorporated into the new heading."
     (org-roam-erase-keyword "title")
     (org-roam-erase-keyword "filetags")))
 
+(defun org-roam--heading-levels ()
+  "Extract the heading level structure from the current file."
+  (org-map-entries (lambda () (org-current-level))
```

Comment:
we don't want to be using `org-map-entries` here since it introduces dependencies on `org-agenda` related configuration. Use `org-map-region` istead as per line 848 to 854.